cmd/client-s3.go

```diff
@@ -315,15 +315,75 @@ func (c *S3Client) RemoveNotificationConfig(arn string) *probe.Error {
 	}
 	accountArn := minio.NewArn(fields[1], fields[2], fields[3], fields[4], fields[5])
 
-	switch fields[2] {
-	case "sns":
-		mb.RemoveTopicByArn(accountArn)
-	case "sqs":
-		mb.RemoveQueueByArn(accountArn)
-	case "lambda":
-		mb.RemoveLambdaByArn(accountArn)
-	default:
-		return errInvalidArgument().Trace(fields[2])
+	// if we are passed filters for either events, suffix or prefix, then only delete the single event that matches
+	// the arguments
+	if event != "" || suffix != "" || prefix != "" {
+		// Translate events to type events for comparison
+		events := strings.Split(event, ",")
+		var eventsTyped []minio.NotificationEventType
+		for _, e := range events {
+			switch e {
+			case "put":
+				eventsTyped = append(eventsTyped, minio.ObjectCreatedAll)
+			case "delete":
+				eventsTyped = append(eventsTyped, minio.ObjectRemovedAll)
+			case "get":
+				eventsTyped = append(eventsTyped, minio.ObjectAccessedAll)
+			default:
+				return errInvalidArgument().Trace(events...)
+			}
+		}
+		// based on the arn type, we'll look for the event in the corresponding sublist and delete it if there's a match
+		switch fields[2] {
+		case "sns":
+			removeIndex := -1
+			for i, v := range mb.TopicConfigs {
+				// if it matches events and filters, mark the index for deletion
+				if EqualNotificationConfig(minio.NotificationConfig{Arn: v.Arn, Events: v.Events, Filter: v.Filter}, arn, eventsTyped, prefix, suffix) {
```

Comment:
```suggestion
				if EqualNotificationConfig(v.NotificationConfig, arn, eventsTyped, prefix, suffix) {
```

Reply:
I don't know why I didn't think of that, that is way more convenient, I was craving generics to avoid this 3 blocks of code since this 3 structs share `NotificationConfig `

Reply:
done